rollrus.go

```diff
@@ -2,60 +2,128 @@ package rollrus
 
 import (
 	"fmt"
+	"io"
 	"os"
+	"sync"
 	"time"
+	"runtime"
 
+	"github.com/benjamindow/rollrus/throttle"
 	log "github.com/sirupsen/logrus"
 	"github.com/stvp/roll"
 )
 
+type noopCloser struct{}
+
+func (c noopCloser) Close() error {
+	return nil
+}
+
+type RollrusConfig struct {
+	BufferSize int
+	NumWorkers int
+	LogLevels  []log.Level
+	Throttler  throttle.Throttler
+}
+
 var defaultTriggerLevels = []log.Level{
 	log.ErrorLevel,
 	log.FatalLevel,
 	log.PanicLevel,
 }
 
+var defaultNumWorkers = 8 * runtime.NumCPU()
+var defaultBufferSize = 2 * defaultNumWorkers
+
 // Hook wrapper for the rollbar Client
 // May be used as a rollbar client itself
 type Hook struct {
 	roll.Client
-	triggers []log.Level
+	triggers      []log.Level
+	buffer        chan *log.Entry
```

Comment:
Have considered replacing this with a diode interested to see what people thing

https://github.com/cloudfoundry/go-diodes